packages/next-server/lib/router/router.ts

```diff
@@ -277,7 +277,16 @@ export default class Router implements BaseRouter {
         return resolve(true)
       }
 
-      const { pathname, query } = parse(url, true)
+      const { pathname, query, protocol } = parse(url, true)
+
+      if (!pathname || protocol) {
```

Comment:
this condition looks wrong, IMO should be `!pathname || !protocol`

Reply:
I agree, just stumbled upon this today using `Router.push` and passing a URL object to both `href`, and `as`.